controllers/argocd/sso.go

```diff
@@ -164,7 +165,8 @@ func (r *ReconcileArgoCD) reconcileSSO(cr *argoprojv1a1.ArgoCD) error {
 		if cr.Spec.SSO.Provider == v1alpha1.SSOProviderTypeKeycloak {
 			// Relevant SSO settings at play are `DISABLE_DEX`, `.spec.dex`, `.spec.sso` fields, `.spec.sso.keycloak`, `.spec.sso.dex`
 
-			if (cr.Spec.SSO.Keycloak != nil) && (cr.Spec.SSO.Image != "" || cr.Spec.SSO.Version != "" || cr.Spec.SSO.Resources != nil || cr.Spec.SSO.VerifyTLS != cr.Spec.SSO.Keycloak.VerifyTLS) {
+			if (cr.Spec.SSO.Keycloak != nil) && (cr.Spec.SSO.Image != "" || cr.Spec.SSO.Version != "" ||
```

Comment:
1. Improve the readability of code.
2. The below condition should be updated as it is breaking on the CR.
```
apiVersion: argoproj.io/v1alpha1
kind: ArgoCD
metadata:
  name: example-argocd
  labels:
    example: route
spec:
  sso:
   provider: keycloak
   keycloak:
     verifyTLS: false
  server:
    route:
      enabled: true
```

Reply:
@iam-veeramalla why is it breaking in this case? if `cr.spec.SSO.verifyTLS` is not expressed I think default should be false, that should mean `cr.spec.SSO.verifyTLS` and `cr.spec.SSO.keycloak.verifyTLS` should match and it should not error out

Reply:
@jaideep, if `cr.spec.SSO.verifyTLS` is not expressed, it will default to `true` not `false` as it is not a boolean it references a boolean.

Reply:
I don't think it will have a boolean value at all, since it is a pointer, it will just be `nil`.  Both `cr.spec.SSO.verifyTLS` and `cr.spec.SSO.verifyTLS` are pointers to boolean, so checking if they have different values is going to be more complex.  Basically they are different if one is set and the other is not, or one points to true and the other points to false.

Reply:
My comment above is irrelevant.  The change Abhishek made is correct, since `cr.spec.SSO.verifyTLS` being a pointer means it should be nil if keycloak is configured through `cr.spec.sso.keycloak`.